intf/accessor_linux.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package intf
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/vishvananda/netlink"
+)
+
+func init() {
+	// On init when the package has been built on Linux, load the netlink
+	// accessor as the implementation to be used.
+	//
+	// TODO(robjs): Consider whether we should use build tags vs. solely
+	// the underlying system, since this would allow a build tag to say
+	// that the package should use gRPC wire.
+	accessor = netlinkAccessor{}
```

Comment:
defaultAccessor ?

Reply:
My logic here was that the `_linux.go` file only gets built if the build target is Linux, so it's only really the 'default' if you're on a Linux system.

Going forward, with gRPCwire, I expect we need a similar thing that interfaces to e.g., read from an ARP cache, add interface IPs etc. from a caller, so I suspect that is then done with a different build tag. Does that pattern make sense?